Firestore/Swift/Source/PropertyWrapper/FirestoreQueryObservable.swift

```diff
@@ -19,7 +19,7 @@ import FirebaseFirestore
 
 @available(iOS 14.0, macOS 11.0, macCatalyst 14.0, tvOS 14.0, watchOS 7.0, *)
 internal class FirestoreQueryObservable<T>: ObservableObject {
-  @Published var items: T
+  @Published var result: T
```

Comment:
I am somewhat new to this subject, so apologies in advance for potential irrelevance, , but wouldn't most developers expect that any api's that currently return a Result object will be deprecated in favour of new api's using async /await since that also allows direct error trapping  ?  Obviously not immediately, the old API should stay for a while, as long as it can sit nicely alongside easier/newer API's.

Reply:
As of now property wrappers can't throw, so a/a isn't really applicable I believe. Is there something that I'm missing?

Reply:
Currently, that is correct, however in the proposal [SE-310](https://github.com/apple/swift-evolution/blob/main/proposals/0310-effectful-readonly-properties.md) indicates that at a future point in time, we may have throwing capability from property wrappers. You can currently throw from an async computed get accessor, and since a property wrapper is mostly syntactic sugar, it shouldn't be too hard to accomplish?. One can hope!

Reply:
> SE-310 indicates that at a future point in time, we may have throwing capability from property wrappers

We should definitely keep an eye on this. However, as this isn't available yet, let's stick to what we can use now.

Reply:
Just a thought, would it be possible to return an object from a function call that gives us the required behaviour?
e.g. var fruits: FirestoreQuery
do { try await fruits = FirestoreQuery(query: <String>) }
catch {  /* Error handling */ }
..
then allow iteration over fruits.wrappedValue, and access the projected value for other object details?

I haven't thought that through, but just throwing an idea out there  :p